e2e/nomostest/client.go

```diff
@@ -93,13 +94,23 @@ func newScheme(t testing.NTB) *runtime.Scheme {
 // RestConfig sets up the config for creating a Client connection to a K8s cluster.
 // If --test-cluster=kind, it creates a Kind cluster.
 // If --test-cluster=kubeconfig, it uses the context specified in kubeconfig.
-func RestConfig(t testing.NTB, optsStruct *ntopts.New) {
+func RestConfig(t testing.NTB, opts *ntopts.New) {
 	switch strings.ToLower(*e2e.TestCluster) {
 	case e2e.Kind:
-		ntopts.Kind(t, *e2e.KubernetesVersion)(optsStruct)
+		ntopts.Kind(t, *e2e.KubernetesVersion)(opts)
 	case e2e.GKE:
-		ntopts.GKECluster(t, restconfig.DefaultTimeout)(optsStruct)
+		ntopts.GKECluster(t, restconfig.DefaultTimeout)(opts)
 	default:
 		t.Fatalf("unsupported test cluster config %s. Allowed values are %s and %s.", *e2e.TestCluster, e2e.GKE, e2e.Kind)
 	}
+
+	restConfig, err := restconfig.NewFromConfigFile(opts.KubeconfigPath)
+	if err != nil {
+		t.Fatalf("building rest.Config: %v", err)
+	}
+	opts.RESTConfig = restConfig
+
+	// Copy the RESTConfig to create the WatchConfig, but use a longer timeout.
+	opts.WatchConfig = restconfig.DeepCopy(restConfig)
+	opts.WatchConfig.Timeout = watch.RESTConfigTimeout
```

Comment:
Is it important we use the same timeout defined in remediator, or can we just define our own timeout here?

Reply:
It's not important, but the constant existed already. It's basically just "a really long time".

When using an Informer, it handles restarting cancelled requests, so it's not really a problem as long as it's long enough to not be creating new requests all the time.

However, when using a raw Watch, like in WatchForObject, the timeout just needs to be longer than the timeout configured in the WaitOption (DefaultWaitTimeout by default).